lib/linters/mail_later_linter.rb

```diff
@@ -11,14 +11,27 @@ module IdentityIdp
       #
       #   #good
       #   UserMailer.signup_with_your_email(user, email).deliver_now_or_later
+      #   UserMailer.with(params).signup_with_your_email(user, email).deliver_now_or_later
+      #   ReportMailer.report_mail(data).deliver_now
       #
       class MailLaterLinter < RuboCop::Cop::Cop
         MSG = 'Please send mail using deliver_now_or_later instead'.freeze
 
         RESTRICT_ON_SEND = [:deliver_now, :deliver_later].freeze
 
         def on_send(node)
-          add_offense(node, location: :expression)
+          receiver = node.children.first&.receiver
+          return if !receiver
+
+          mailer_name = if receiver.const_type?
+                          # MailerClass.email.send_later
+                          receiver.const_name
+                        elsif receiver.method_name == :with
+                          # MailerClass.with(...).email.send_later
+                          receiver.receiver.const_name
+                        end
```

Comment:
another approach would be to get smarter about matching expressions... something like...

```
(send
  (send
    (send
      (const nil :UserMailer) :with
      $...) $...) :deliver_now)
```
where we have a regex capture for `:deliver_now` and also check `:deliver_later`

or maybe two regexes?

anyways this was something I could brute-force via specs